mongo/dataset_store.go

```diff
@@ -114,20 +135,33 @@ func (m *Mongo) GetDataset(id string) (*models.DatasetUpdate, error) {
 }
 
 // GetEditions retrieves all edition documents for a dataset
-func (m *Mongo) GetEditions(ctx context.Context, id, state string) (*models.EditionUpdateResults, error) {
+func (m *Mongo) GetEditions(ctx context.Context, id, state string, offset, limit int, authorised bool) (*models.EditionUpdateResults, error) {
 	s := m.Session.Copy()
 	defer s.Close()
 
 	selector := buildEditionsQuery(id, state)
 
-	iter := s.DB(m.Database).C(editionsCollection).Find(selector).Iter()
+	var q *mgo.Query
+
+	if authorised {
+		q = s.DB(m.Database).C(editionsCollection).Find(selector)
+	} else {
+		q = s.DB(m.Database).C(editionsCollection).Find(bson.M{"current": bson.M{"$exists": true}})
+	}
+
+	iter := q.Sort().Skip(offset).Limit(limit).Iter()
 	defer func() {
 		err := iter.Close()
 		if err != nil {
 			log.Event(ctx, "error closing edition iterator", log.ERROR, log.Error(err), log.Data{"selector": selector})
 		}
 	}()
 
+	totalCount, err := q.Count()
```

Comment:
Think about the order in which you are making query calls to mongo. I would advise doing the count query first and handling the error like you have done for `GetDatasets`